src/coreclr/jit/codegenlinear.cpp

```diff
@@ -937,9 +937,20 @@ void CodeGen::genSpillVar(GenTree* tree)
         // but we will kill the var in the reg (below).
         if (!varDsc->IsAlwaysAliveInMemory())
         {
-            instruction storeIns = ins_Store(lclType, compiler->isSIMDTypeLocalAligned(varNum));
             assert(varDsc->GetRegNum() == tree->GetRegNum());
-            inst_TT_RV(storeIns, size, tree, tree->GetRegNum());
+#if defined(FEATURE_SIMD)
+            if ((lclType == TYP_SIMD12) && !compiler->lvaMapSimd12ToSimd16(varDsc))
+            {
+                // Store SIMD12 to stack as 12 bytes
+                GetEmitter()->emitStoreSimd12ToLclOffset(varNum, tree->AsLclVarCommon()->GetLclOffs(),
+                                                         tree->GetRegNum(), nullptr);
+            }
+            else
+#endif
+            {
+                instruction storeIns = ins_Store(lclType, compiler->isSIMDTypeLocalAligned(varNum));
+                inst_TT_RV(storeIns, size, tree, tree->GetRegNum());
+            }
```

Comment:
I realize I suggested it like this, but I think we should make this more similar to existing IR semantics, where lowering lowers `TYP_SIMD12` to `TYP_SIMD16` typed nodes when appropriate, and otherwise leaves `TYP_SIMD12` to always mean "12 byte load/store". 

That is, I think you should switch LSRA to call `WidenSIMD12IfNecessary` from inside `LinearScan::insertMove` and then change the code here to make the determination only based on `tree->TypeIs(TYP_SIMD12)`.

Reply:
Done